Jamie/3Sum.java

```diff
@@ -0,0 +1,50 @@
+//3Sum
+public List<List<Integer>> threeSum(int[] nums) {
+        List<List<Integer>> result = new ArrayList<List<Integer>>();
+
+        if (nums.length < 3)
+            return result;
+
+        // sort array
+        Arrays.sort(nums);
+
+        for (int i = 0; i < nums.length - 2; i++) {
+            //avoid duplicate solutions
+            if (i == 0 || nums[i] > nums[i - 1]) {
+
+                int negate = -nums[i];
+
+                int start = i + 1;
+                int end = nums.length - 1;
+
+                while (start < end) {
+                    //case 1
+                    if (nums[start] + nums[end] == negate) {
+                        List<Integer> temp = new ArrayList<Integer>();
+                        temp.add(nums[i]);
+                        temp.add(nums[start]);
+                        temp.add(nums[end]);
+
+                        result.add(temp);
```

Comment:
We can also use the sentence below to check duplicate:
if (!result.contains(temp)) {
    result.add(temp);
}